### PR TITLE
Lots of refactoring to make it work in production

### PR DESCRIPTION
**Pull request by MaxTheComputerer**
Tidied lots of code, refactored stuff to work multi-threaded, added a setup script, etc.

### CODE REVIEW

File: chat_bot/prompts.py [hunk 1]
```diff
@@ -7,22 +7,22 @@
...
-        "food_waste": "This is a conversation between two friends about food and the environment.\n\nFriend_q: What can we do about it food waste?\nYou: We can redistribute food to soup kitchens and food banks. Even restaurants have a lot of waste.\nFriend_q: What can we do about that?\nYou: We can ask them to donate their leftover food too. Why is there so much food waste on our planet?\nFriend_a: We have so much food, but we don't eat it all.\nYou: How come?\nFriend_a: Supermarkets have so much spare food they don't sell."
+        "food waste": "This is a conversation between two friends about food and the environment.\n\nFriend_q: What can we do about it food waste?\nYou: We can redistribute food to soup kitchens and food banks. Even restaurants have a lot of waste.\nFriend_q: What can we do about that?\nYou: We can ask them to donate their leftover food too. Why is there so much food waste on our planet?\nFriend_a: We have so much food, but we don't eat it all.\nYou: How come?\nFriend_a: Supermarkets have so much spare food they don't sell."
```
Comment (meganelisabethfinch) [inline chat_bot/prompts.py]: Typo in Friend Q's first response is annoying me: "What can we do about it food waste?"

File: chat_bot/prompts.py [hunk 2]
```diff
@@ -7,22 +7,22 @@
...
+        "extinction" : "This is a conversation between two friends about endangered species.\n\nFriend_q: Did you hear that the dinosaurs went extinct?\nYou: I know, it's so sad. They were such cool animals.\nFriend_q: It's really sad that so many animals are in danger of becoming extinct. What can we do to help protect them?\nYou: We can protect the environment, and conserve forests and land, to conserve their habitats. What animals are in danger?\nFriend_a: There are so many animals that are in danger of becoming extinct. Some of the animals that are in the most danger are the Tigers, Pandas, and the Gorillas.\nYou: That's so sad. How can we help?\nFriend_a: We can create awareness about the importance of these animals, and the danger they are in."
```
Comment (meganelisabethfinch) [inline chat_bot/prompts.py:15]: I too am sad that dinosaurs went extinct... :(

File: app.py [hunk 3]
```diff
@@ -63,56 +67,80 @@ def about():
...
-    files = glob.glob('temp/*')
-    for f in files:
-        os.remove(f)
-
 @app.route('/upload', methods=['GET', 'POST'])
 def upload():
     if request.method == 'POST':
+        # Save audio file from POST data
         f = request.files['audio_data']
         f.save('temp/upload.wav')
```
Comment (meganelisabethfinch) [inline app.py]: Will the temp file name upload.wav conflict if two users upload or record concurrently? (maybe a bit optimistic of me to assume we get two concurrent users...)

File: app.py [hunk 4]
```diff
@@ -63,56 +67,80 @@ def about():
...
+
+        # Re-encode audio file ready for transcription
         data, samplerate = sf.read('temp/upload.wav')
```
Comment (meganelisabethfinch) [inline app.py]: I will look into doing this without so many writes/reads to file! :D

File: app.py [hunk 5]
```diff
@@ -63,56 +67,80 @@ def about():
...
 def initial_prompt():
     if request.method == 'POST':
-        prompt = "What is your opinion of "+app_data["topic"]+"?"
-        chatbot.prompt += f"\nFriend_q: " + prompt
-        language = lang.tts_langs()[session["language-code"][:2]]
-        translated_p = chatbot.translate(prompt, "English", language)
-        response_dict = {'ai': translated_p}
+        # Start with a basic question as the initial prompt to the user, append this to the GPT-3 hidden prompt
+        start_prompt = f"What is your opinion of {session['subtopic']}?"
+        session["prompt"] += f"\nFriend_q: " + start_prompt
```
Comment (meganelisabethfinch) [inline app.py:108]: Three things:
- Nitpick, but initial prompt is clunky for some topics or subtopics e.g. "What is your opinion of [environment]?"
- FYI initial prompt is also still broken (but only sometimes)
- You might enjoy this prompt Simon just gave me on the topic of environment: "I think extinction is a natural process that will happen to all living species."

**Review by meganelisabethfinch — APPROVED** [review]

File: chat_bot/prompts.py (hunk 2 above)
Comment (MaxTheComputerer) [reply chat_bot/prompts.py:15]: @anoushkamazumdar this was so funny

File: app.py (hunk 3 above)
Comment (MaxTheComputerer) [reply app.py]: Yeah but I can't think of an easy fix so we can worry about it in the future

File: app.py (hunk 5 above)
Comment (MaxTheComputerer) [reply app.py:108]: Initial prompt now works slightly more reliably

**Pull request merged**